src/mozilla_sec_eia/models/sec10k/sec_output_table.py

```diff
@@ -0,0 +1,367 @@
+)
+
+logger = logging.getLogger(f"catalystcoop.{__name__}")
+
+INVALID_NAMES = [
+    "llc",
+    "limited liability company",
+    "limited",
+    "ltd",
+    "iiii",
+    "inc",
+    "incorporated",
+    "partnership",
+    "i",
+    "name",
+    "company",
+    "&",
+    "",
+]
+
+
+def _remove_weird_sec_cols(sec_df) -> pd.DataFrame:
+    weird_cols = ["]fiscal_year_end", "]irs_number", "]state_of_incorporation"]
+    for weird_col in weird_cols:
+        if weird_col not in sec_df:
+            continue
+        normal_col = weird_col[1:]
+        sec_df.loc[:, normal_col] = sec_df[normal_col].where(
+            sec_df[weird_col].isnull(), sec_df[weird_col]
+        )
+        sec_df = sec_df.drop(columns=[weird_col])
+    return sec_df
+
+
+def _add_report_year_to_sec(sec_df: pd.DataFrame, md: pd.DataFrame) -> pd.DataFrame:
+    """Merge metadata on to get a report year for extracted SEC data.
+
+    Expects filename to be the index of the SEC dataframe.
+    """
+    sec_df = sec_df.merge(md[["filename", "date_filed"]], how="left", on=["filename"])
+    sec_df = sec_df.rename(columns={"date_filed": "report_date"})
+    sec_df.loc[:, "report_year"] = (
+        sec_df["report_date"].astype("datetime64[ns]").dt.year
+    )
+    return sec_df
+
+
+def _flatten_sec_companies_across_time(sec_df: pd.DataFrame) -> pd.DataFrame:
+    """Keep only the most recent record for each unique SEC company.
+
+    Note that this drops old records for companies that have changed
+    names or addresses across time. Also, we group by sec_company_id not
+    CIK, so filer companies and subsidiary companies are unique in the
+    output dataframe.
+    TODO: create an asset that tracks name and address chnages across
+    time.
+    """
+    sec_df = (
+        sec_df.sort_values(by="report_year", ascending=False)
+        .groupby("sec_company_id")
+        .first()
+    ).reset_index()
+    return sec_df
+
+
+def get_sec_state_code_dict() -> dict[str, str]:
+    """Create a dictionary mapping state codes to their names.
+
+    Table found at https://www.sec.gov/submit-filings/filer-support-resources/edgar-state-country-codes
+    Published by SEC and reports valid state codes
+    for filers of Form D. Used to standardize the state codes
+    in the SEC 10K filings. The expanded names of the state codes
+    are comments in the XML file, so we have to read the XML in as
+    text and parse it.
+    """
+    # TODO: make a check to see if SEC has published a new version of this table
+    xml_filepath = (
+        resources.files("mozilla_sec_eia.package_data") / "formDStateCodes.xsd.xml"
+    )
+    with Path.open(xml_filepath) as file:
+        xml_text = file.read()
+
+    pattern = r'<xs:enumeration value="(.*?)"/>.*?<!--\s*(.*?)\s*-->'
+    state_code_dict = {
+        code.lower(): name.lower()
+        for code, name in re.findall(pattern, xml_text, re.DOTALL)
+    }
+    return state_code_dict
+
+
+def clean_location_of_inc(df) -> pd.DataFrame:
```

Comment:
These location of incorporation and company name cleaning functions could maybe be moved to a different module?